jekyll-assets/_includes/theme.html

```diff
@@ -0,0 +1,3 @@
+<div id="theme-toggle-container">
+	<a id="theme-toggle" alt="toggle site theme" title="toggle site theme" onclick="toggleTheme()"></a>
```

Comment:
Would it be better to be unobtrusive here and [add a `click` event listener](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener) on the element (identifying it by its ID of `theme-toggle`)?

e.g.

```javascript
document.getElementById("theme-toggle").addEventListener("click", (e) => {
  e.preventDefault();

  toggleTheme();
});
```

Reply:
My understanding is that this ultimately does the exact same thing, but with an element lookup. Is there a performance benefit that I'm not aware of, or some penalty to using onclick?

Reply:
See [my other comment](https://github.com/raspberrypi/documentation/pull/3950#discussion_r1868086493) re separating the HTML from the JavaScript but also the broader concept of [unobtrusive JavaScript](https://en.wikipedia.org/wiki/Unobtrusive_JavaScript) to encourage [progressive enhancement](https://www.gov.uk/service-manual/technology/using-progressive-enhancement).